lib/ansible/modules/cloud/podman/podman_container_info.py

```diff
@@ -0,0 +1,410 @@
+                        0,
+                        0
+                    ],
+                    "Isolation": "",
+                    "CpuShares": 0,
+                    "Memory": 0,
+                    "NanoCpus": 0,
+                    "CgroupParent": "",
+                    "BlkioWeight": 0,
+                    "BlkioWeightDevice": null,
+                    "BlkioDeviceReadBps": null,
+                    "BlkioDeviceWriteBps": null,
+                    "BlkioDeviceReadIOps": null,
+                    "BlkioDeviceWriteIOps": null,
+                    "CpuPeriod": 0,
+                    "CpuQuota": 0,
+                    "CpuRealtimePeriod": 0,
+                    "CpuRealtimeRuntime": 0,
+                    "CpusetCpus": "",
+                    "CpusetMems": "",
+                    "Devices": [],
+                    "DiskQuota": 0,
+                    "KernelMemory": 0,
+                    "MemoryReservation": 0,
+                    "MemorySwap": 0,
+                    "MemorySwappiness": -1,
+                    "OomKillDisable": false,
+                    "PidsLimit": 0,
+                    "Ulimits": [
+                        {
+                            "Name": "RLIMIT_NOFILE",
+                            "Soft": 1048576,
+                            "Hard": 1048576
+                        },
+                        {
+                            "Name": "RLIMIT_NPROC",
+                            "Soft": 1048576,
+                            "Hard": 1048576
+                        }
+                    ],
+                    "CpuCount": 0,
+                    "CpuPercent": 0,
+                    "IOMaximumIOps": 0,
+                    "IOMaximumBandwidth": 0
+                }
+                }
+        ]
+"""
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+def get_containers_facts(module, executable, name):
+    """Collect containers facts for all containers or for specified in 'name'.
+
+    Arguments:
+        module {AnsibleModule} -- instance of AnsibleModule
+        executable {string} -- binary to execute when inspecting containers
+        name {list} -- list of names or None in case of all containers
+
+    Returns:
+        list of containers info, stdout, stderr
+    """
+    if not name:
+        all_names = [executable, 'container', 'ls', '-q', '-a']
+        rc, out, err = module.run_command(all_names)
+        if rc != 0:
+            module.fail_json(msg="Unable to get list of containers: %s" % err)
+        name = out.split()
+        if not name:
+            return [], out, err
+    command = [executable, 'container', 'inspect']
+    command.extend(name)
+    rc, out, err = module.run_command(command)
+    if rc == 0:
+        json_out = json.loads(out) if out else None
+        if json_out is None:
+            return [], out, err
+        return json_out, out, err
+    if rc != 0 and 'no such ' in err:
+        if len(name) < 2:
```

Comment:
What's the point of this test?

Reply:
When you do multiple inspection in podman and one containers is missing it will fail. If we have only one container to inspect - it means we don't have anything to work on. If we have multiple container, it could be that one missing only and we can go "cycle_over" function below.